lib/iris/analysis/_interpolation.py

```diff
@@ -296,44 +312,51 @@ def _interpolate(self, data, interp_points):
             coordinate, which is mapped to the i'th data dimension.
             The other (leading) dimensions index over the different required
             sample points.
+        interp_shape :
+            The shape of the interpolated array in its first 'N' dimensions
+            (len(interp_shape) should be N).
+        method: str
+            Interpolation method (see :class:`iris.analysis._interpolation.RectilinearInterpolator`)
+        extrapolation_mode: str
+            Extrapolation mode (see :class:`iris.analysis._interpolation.RectilinearInterpolator`)
 
         Returns
         -------
         :class:`np.ndarray`.
-            Its shape is "points_shape + extra_shape",
+            Its shape is "interp_shape + extra_shape",
             where "extra_shape" is the remaining non-interpolated dimensions of
-            the data array (i.e. 'data.shape[N:]'), and "points_shape" is the
-            leading dimensions of interp_points,
-            (i.e. 'interp_points.shape[:-1]').
-
+            the data array (i.e. 'data.shape[N:]').
         """
-        from iris.analysis._scipy_interpolate import _RegularGridInterpolator
-
-        dtype = self._interpolated_dtype(data.dtype)
+        dtype = _interpolated_dtype(data.dtype, method)
         if data.dtype != dtype:
             # Perform dtype promotion.
             data = data.astype(dtype)
 
-        mode = EXTRAPOLATION_MODES[self._mode]
-        if self._interpolator is None:
-            # Cache the interpolator instance.
-            # NB. The constructor of the _RegularGridInterpolator class does
-            # some unnecessary checks on the fill_value parameter,
-            # so we set it afterwards instead. Sneaky. ;-)
-            self._interpolator = _RegularGridInterpolator(
-                self._src_points,
-                data,
-                method=self.method,
-                bounds_error=mode.bounds_error,
-                fill_value=None,
-            )
-        else:
-            self._interpolator.values = data
```

Comment:
`_interpolate` is now mapped to all the chunks of the data we are interpolating, thus we cannot cache the `_RegularGridInterpolator` instance anymore. Is this acceptable? 

One can still cache the  `RectilinearInterpolator` thought, and, as far as I can say, this would be similar to the way in which the caching of the regridder works?

Reply:
Having read through the code base: I believe the caching is because the same interpolation routine gets used on a series of coordinates - start here and follow the call stack down to `_points()`:

https://github.com/SciTools/iris/blob/4c73724b2bad6469c2b75e7ed54d5be8195d8988/lib/iris/analysis/_interpolation.py#L656-L658

... meaning multiple uses of `_RegularGridInterpolator` but with only `.values` changing. I assume that there is noticeable overhead when creating a new instance of `_RegularGridInterpolator`, but that there is no problem applying the same instance to a series of different arrays, hence the caching.

There is a risk that your refactoring - which makes the routine much faster when parallelising large datasets - will make small-to-medium cases slower, especially for those cases that need to interpolate a larger number of coordinates. So I am very keen that we try out my suggestion using `args=[self]`, which I believe would allow us to retain the existing caching? If that does not work then I'll need to assess the performance impact with some benchmarking.